homeassistant/components/esphome/diagnostics.py

```diff
@@ -0,0 +1,33 @@
+"""Diahgnostics support for ESPHome."""
+from __future__ import annotations
+
+from typing import Any, cast
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+
+from . import CONF_NOISE_PSK, DomainData
+
+CONF_MAC_ADDRESS = "mac_address"
+
+REDACT_CONFIG = {CONF_NOISE_PSK, CONF_PASSWORD}
+REDACT_STORAGE_INFO = {CONF_MAC_ADDRESS}
```

Comment:
`async_redact_data` is recursive so you can combine these and do a single call on `diag` at the end

Reply:
I saw it was recursive, but did not think to use it on the whole `diag` :man_facepalming: